### DISCUSSION TITLE
Concatenating two Arrays

### DISCUSSION THREAD

**Discussion by alv-around** [mhostetter/galois#572] [Q&A] [answered]
First of all thanks for the project. I think is a very cool idea, but I am struggling in with a very simple case:

## Goal: I want to concatenate to FieldArrays
I have two arrays $A, B \in F_{p}^{nxn}$ and I would like to obtain $C \in F_{p}^{2nxn}$ (or $F_{p}^{nx2n}$, the axis in my case is not important). 

Put it in code:
```python
A = F([[1, 2], [3, 4]])
B = F([[5, 6], [7, 8]])
C = F([[1, 2], [3, 4], [5, 6], [7, 8]])
```

## Failed Approaches:
I have tried a couple approaches before comming here, and all have failed:
- `np.concatenate(A, B)` results in the following error:
```console
    C = np.concatenate(A, B)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alv/.local/share/virtualenvs/binius-uW3OI3G1/lib/python3.12/site-pack
ages/galois/_domains/_function.py", line 373, in __array_function__
    output = super().__array_function__(func, types, args, kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: only integer scalar arrays can be converted to a scalar index
```
- `A.append(B)`I get the following error:
```console
     C = A.append(B)
              ^^^^^^^^^^^^
AttributeError: 'FieldArray_101_2' object has no attribute 'append'
```
- `A + B` -> The "pure" python way, which results in C being an Array of the same size as A and B and provides element-wise sum. That is totally fine and is aligned with np. I just added here to be exhaustive.

 I am missing something?
 
  If I could go out of my way, to get it work. But I also think that if the purpose of the library is to provide a seamless experience with np, scipy, ... such an elementary operation should provide a simple interface.

## Solution

After a couple more trials I made it work by calling:
```python
C = np.append(A, B, axis=0)
```

**Comment by mhostetter** [comment, marked as answer]
Is this your goal? I think `np.vstack()` and `np.hstack()` may be what you're looking for.

```python
import galois

F = galois.GF(101)

A = F([[1, 2], [3, 4]])
B = F([[5, 6], [7, 8]])
C = np.vstack((A, B))

print(C)
```

```
[[1 2]
 [3 4]
 [5 6]
 [7 8]]
```

**Reply by alv-around** [reply]
yes! Thank you very much :)
